tests/test_dev_server.py

```diff
@@ -49,10 +48,87 @@ def test_my_lambda_dev_handler():
     assert path == "/t/{id}"
     assert params == {"id": "123"}
 
+    path, params = handler._get_route_params("/?=1&xx=2")  # pylint: disable=protected-access
+    assert path == "/"
+    assert params is None
+
+    with pytest.raises(ValueError):
+        handler._get_route_params("")  # pylint: disable=protected-access
+
+
+@pytest.mark.parametrize(
+    "method", ["do_GET", "do_PATCH", "do_POST", "do_PUT", "do_DELETE", "do_OPTIONS"]
+)
+def test_handle_request(method) -> None:
+    with patch("socket.socket") as msocket:
+        msocket.makefile = lambda a, b: io.BytesIO(b"GET / HTTP/1.1\r\n")
+        msocket.rfile.close = lambda: 0
```

Comment:
The `msocket` variable should become a separate fixture. Mostly, to describe that better but also, to reduce the duplication. The biggest value of the tests is the documentation which can be checked anytime by anyone - so, we need to try to make the tests simple 😉